bin/benchmark/Sources/benchmark/Diff/DiffAnalysis.swift

```diff
@@ -0,0 +1,219 @@
+            change = .differentNumeric(percentage: approximateChange)
+            
+            footnotes = [
+                .init(text: "The before and after values are different enough that the most probable explanation is that random samples from two different data sets.", values: footnoteValues)
+            ]
+        }
+        
+        if !warnings.isEmpty {
+            footnotes.append(
+                .init(text: "A human should check that the measurements for this metric look like random samples around a certain value to ensure the validity of this result.", values: [])
+            )
+        }
+        
+        return MetricAnalysis(
+            metricName: afterMetric.displayName,
+            metricID: afterMetric.id,
+            change: change,
+            before: beforeMetric.values.formatted(),
+            after: afterMetric.values.formatted(),
+            footnotes: footnotes,
+            warnings: warnings.isEmpty ? nil : warnings
+        )
+    }
+    
+    private static func inputBiasDescription(metric: BenchmarkResultSeries.MetricSeries, sampleName: String, numbers: [Double]) -> String {
+        // Turn the single metric series into an array of single values metric series to render the trend bars.
+        
+        let metricSeries: [BenchmarkResultSeries.MetricSeries]
+        switch metric.values {
+            case .duration(let values):
+                metricSeries = values.map {
+                    .init(id: metric.id, displayName: metric.displayName, values: .duration([$0]))
+                }
+                
+            case .bytesInMemory(let values):
+                metricSeries = values.map {
+                    .init(id: metric.id, displayName: metric.displayName, values: .bytesInMemory([$0]))
+                }
+            case .bytesOnDisk(let values):
+                metricSeries = values.map {
+                    .init(id: metric.id, displayName: metric.displayName, values: .bytesOnDisk([$0]))
+                }
+                
+            case .checksum:
+                // Only numeric metrics support trend bars
+                fatalError("Can't compute input bias for checksum metrics.")
+        }
+        
+        return """
+        The '\(metric.id)' samples from the '\(sampleName)' measurements show _possible_ signs of bias.
+        The benchmark diff analysis assumes that these values follow a normal distribution around the value that they're meant to represent. If they don't then the conclusion from the diff analysis could be invalid.
+        A human should inspect these values. If the samples look biased or look too noisy you can try gathering new metrics with more 'repetitions' or with a larger .docc catalog.
+        [\(numbers.map { warningNumberFormatter.string(from: NSNumber(value: $0))! }.joined(separator: ", "))]
+        \(RenderTrendAction.renderTrendBars(metrics: metricSeries))
```

Comment:
I'm not sure sure I understand why the numbers displayed in the array above are different to the ones in the trend bars. Is there a reason for that?

Reply:
I addressed this in c34eaaa078ea63733969ca2e21f5449bf7d336ef by printing "full precision values: " before the numerical values and dimming that line to deemphasize it.